### PR TITLE
MagicSearch: blur input on empty search term.

### PR DESCRIPTION
**Pull request by budzanowski**
### Info 
This PR changes the `Enter` key behavior in `MagicSearch`. Now if previous char was space and user clicked `Enter` the `Search` will loose focus. 

This feature was requested in #12266 
@folletto I was wondering if maybe we should expand this to loos focus on `Enter` when nothing has changed?

### Testing
Open `/design` and start typing into Input. When suggestion picked with `Enter` focus is not lost. When `Enter` is used and the previous char is different than space focus is not lost. When previous char is space and `Enter` is used focus is lost.

### CODE REVIEW

**Comment by folletto** [comment]
Tested in Chrome, works 👍

File: client/my-sites/themes/themes-magic-search-card/index.jsx [hunk 1]
```diff
@@ -57,13 +57,25 @@ class ThemesMagicSearchCard extends React.Component {
...
+		this.findTextForSuggestions( txt );
 
+		let inputUpdated = false;
 		//We need this logic because we are working togheter with different modules.
 		//that provide suggestions to the input depending on what is currently in input
 		const target = this.state.editedSearchElement !== '' ? 'suggestions' : 'welcome';
 		if ( this.refs[ target ] ) {
-			this.refs[ target ].handleKeyEvent( event );
+			// handleKeyEvent functions return bool that infroms if suggestion was picked
+			// We need that because we cannot relly on input state because it is updated
```
Comment (folletto) [inline client/my-sites/themes/themes-magic-search-card/index.jsx]: Typo: "relly" → "rely"

File: client/my-sites/themes/themes-magic-search-card/welcome.jsx [hunk 2]
```diff
@@ -56,9 +56,11 @@ class MagicSearchWelcome extends React.Component {
 					this.props.suggestionsCallback( this.props.taxonomies[ position ] + ':' );
 					event.stopPropagation();
 					event.preventDefault();
+					return true;
```
Comment (seear) [inline client/my-sites/themes/themes-magic-search-card/welcome.jsx]: Again, we should add jsdoc for this function.

File: client/my-sites/themes/themes-magic-search-card/index.jsx [hunk 3]
```diff
@@ -57,13 +57,25 @@ class ThemesMagicSearchCard extends React.Component {
...
+			// asynchronously and we are not able to observe what was changed during handleKeyEvent
+			inputUpdated = this.refs[ target ].handleKeyEvent( event );
+		}
+
+		if ( event.key === 'Enter' && ! inputUpdated ) {
+			const cursorPosition = txt.slice( 0, this.refs[ 'url-search' ].refs.searchInput.selectionStart ).length;
+			if ( txt[ cursorPosition - 1 ] === ' ' ) {
```
Comment (seear) [inline client/my-sites/themes/themes-magic-search-card/index.jsx]: Maybe we should pull these couple of lines out as a function, for example `isPreviousCharWhitespace()`, chiefly to document what is going on?

Not a blocker, though.

**Pull request merged**